ts/features/bonus/siciliaVola/components/CheckIncomeComponent.tsx

```diff
@@ -0,0 +1,119 @@
+import { GlobalState } from "../../../../store/reducers/types";
+import {
+  svGenerateVoucherBack,
+  svGenerateVoucherCancel,
+  svGenerateVoucherUnderThresholdIncome
+} from "../store/actions/voucherGeneration";
+import FooterWithButtons from "../../../../components/ui/FooterWithButtons";
+import { navigateToSvKoCheckIncomeThresholdScreen } from "../navigation/actions";
+import I18n from "../../../../i18n";
+import {
+  RadioButtonList,
+  RadioItem
+} from "../../../../components/core/selection/RadioButtonList";
+
+type OwnProps = {
+  onContinuePress: () => void;
+};
+type Props = OwnProps &
+  ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+const getCheckIncomeUnderThresholItems = (): ReadonlyArray<
+  RadioItem<boolean>
+> => [
+  {
+    label: I18n.t("bonus.sv.voucherGeneration.checkIncome.threshold.under"),
+    id: true
+  },
+  {
+    label: I18n.t("bonus.sv.voucherGeneration.checkIncome.threshold.over"),
+    id: false
+  }
+];
+
+const CheckIncomeComponent = (props: Props): React.ReactElement => {
+  const [incomeUnderThreshold, setIncomeUnderThreshold] = React.useState<
+    boolean | undefined
+  >();
+
+  const elementRef = useRef(null);
+
+  const handleContinue = () => {
+    if (incomeUnderThreshold !== undefined) {
+      props.underThresholdIncome(incomeUnderThreshold);
+    }
+    if (incomeUnderThreshold === true) {
+      props.onContinuePress();
+    } else {
+      props.navigateToSvKoCheckIncomeThreshold();
+    }
+  };
+
+  const cancelButtonProps = {
+    primary: false,
+    bordered: true,
+    onPress: props.cancel,
+    title: I18n.t("global.buttons.cancel")
+  };
+  const continueButtonProps = {
+    bordered: false,
+    onPress: handleContinue,
+    title: I18n.t("global.buttons.continue"),
+    disabled: incomeUnderThreshold === undefined
+  };
+
+  return (
+    <BaseScreenComponent
+      goBack={true}
+      contextualHelp={emptyContextualHelp}
+      headerTitle={I18n.t("bonus.sv.headerTitle")}
+    >
+      <SafeAreaView
+        style={IOStyles.flex}
+        testID={"CheckIncomeComponent"}
+        ref={elementRef}
+      >
+        <ScrollView style={[IOStyles.horizontalContentPadding]}>
+          <H1>{I18n.t("bonus.sv.voucherGeneration.checkIncome.title")}</H1>
+          <View spacer={true} />
+          <RadioButtonList<boolean>
+            key="check_income"
+            items={getCheckIncomeUnderThresholItems()}
+            selectedItem={incomeUnderThreshold}
+            onPress={setIncomeUnderThreshold}
+          />
+        </ScrollView>
+        <FooterWithButtons
+          type={"TwoButtonsInlineHalf"}
+          leftButton={cancelButtonProps}
+          rightButton={continueButtonProps}
+        />
+      </SafeAreaView>
+    </BaseScreenComponent>
+  );
+};
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  cancel: () => dispatch(svGenerateVoucherCancel()),
+  back: () => dispatch(svGenerateVoucherBack()),
```

Comment:
```suggestion
  back: () => dispatch(svGenerateVoucherBack()),
```
it seems to be not used

Reply:
It seems right! Removed in [6156f5d](https://github.com/pagopa/io-app/pull/3278/commits/6156f5d1ccc8a7bde96dbc8b1123141bc53c3fcf)